lib/internal/test_runner/test.js

```diff
@@ -788,6 +788,8 @@ class Test extends AsyncResource {
   }
 
   createSubtest(Factory, name, options, fn, overrides) {
+    const timeoutPerTest = this.config.timeoutPerTest;
```

Comment:
The changes in this function seem like they belong in the `Test` constructor. This function doesn't really deal with handling test configuration options. All of that logic is in `Test()`, and there are separate code paths there for the root test vs. all other tests, which is what it looks like you're trying to do here.

Reply:
Thanks for all the reviews, I will address them tomorrow. But before I hit the hay, may I ask real quick just for this one please.

The reason I did the logic here is because it can guarantee that test timeout is set to per test, not on the `Suite` (e.g. `describe` which is the parent) or `FileTest` (which is per file).

I have attempted to move this part into the constructor above and I can get through when `isTestRunner` is false (without `--test` flag). But with `isTestRunner` is true, the timeout is always set on the file level.

Maybe I am missing something here?

Reply:
I would have to check more carefully, but I think the `--test-timeout` logic is probably buggy since the implementation itself is wrong.